From cfe07f4cf4baa449549dea6c536cbac3668bd63b Mon Sep 17 00:00:00 2001
From: Pan Junzhong <junzhong.pan@spacemit.com>
Date: Tue, 16 Apr 2024 13:58:59 +0800
Subject: [PATCH 0599/1206] usb: f_tcm: fix memory leak in
 bot_send_write_request

Remove data_buf allocation and initialization which will be
handled inside usbg_prepare_w_request.

Change-Id: I598ddfbc95ca7c0eb379c7191b0ba94f914c2ace
---
 drivers/usb/gadget/function/f_tcm.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/drivers/usb/gadget/function/f_tcm.c b/drivers/usb/gadget/function/f_tcm.c
index 24a9544dafa4..f7cf77ae9246 100755
--- a/drivers/usb/gadget/function/f_tcm.c
+++ b/drivers/usb/gadget/function/f_tcm.c
@@ -246,7 +246,6 @@ static int bot_send_write_request(struct usbg_cmd *cmd)
 {
 	struct f_uas *fu = cmd->fu;
 	struct se_cmd *se_cmd = &cmd->se_cmd;
-	struct usb_gadget *gadget = fuas_to_gadget(fu);
 	int ret;
 
 	init_completion(&cmd->write_complete);
@@ -257,18 +256,6 @@ static int bot_send_write_request(struct usbg_cmd *cmd)
 		return -EINVAL;
 	}
 
-	if (!gadget->sg_supported) {
-		cmd->data_buf = kmalloc(se_cmd->data_length, GFP_KERNEL);
-		if (!cmd->data_buf)
-			return -ENOMEM;
-
-		fu->bot_req_out->buf = cmd->data_buf;
-	} else {
-		fu->bot_req_out->buf = NULL;
-		fu->bot_req_out->num_sgs = se_cmd->t_data_nents;
-		fu->bot_req_out->sg = se_cmd->t_data_sg;
-	}
-
 	fu->bot_req_out->complete = usbg_data_write_cmpl;
 	fu->bot_req_out->length = se_cmd->data_length;
 	fu->bot_req_out->context = cmd;
-- 
2.47.0

